src/win32/fs.cpp

```diff
@@ -1109,95 +1097,86 @@ size_t WinFileSystemAccess::lastpartlocal(const string* name) const
     return 0;
 }
 
+
 // return lowercased ASCII file extension, including the . separator
 bool WinFileSystemAccess::getextension(const LocalPath& filenamePath, char* extension, size_t size) const
 {
-    const string* filename = filenamePath.editStringDirect();
-
-    const wchar_t* ptr = (const wchar_t*)(filename->data() + filename->size() 
-        - (filename->size() & 1));   // if the string has had an extra null char added for surety, get back on wchar_t boundary.
+    const wchar_t* ptr = filenamePath.localpath.data() + filenamePath.localpath.size();
 
     char c;
     size_t i, j;
 
-	size--;
```

Comment:
From here onward, if only the indentation has changed, then it really shouldn't have.
If implementation changed as well, then it can be massively simplified, using some std functions.

Reply:
I think the original indentation was skewed and not readable. However, there is more than just the indentation. localpath is wstring now, so don't need to multiply with sizeof(wchar_t) at a few places. 
Yes, there are opportunities to use stl and simplify, but I didn't focus on that in particular.

Reply:
already addressed - also removed one cast that's now unnecessary